### PR TITLE
Update to xenial [1.1.1]

### PR DESCRIPTION
**Pull request by bernd-edlinger**
Fix build issues for 1.1.1 branch

### CODE REVIEW

**Comment by levitte** [comment]
Cool.  Do you plan on cherry picking to master as well, or are you making a separate PR for that?

It might be a good idea to have one test build in the matrix that builds on trusty as well, just to make sure.

**Comment by levitte** [comment]
Apart from that, I like what I see

**Comment by bernd-edlinger** [comment]
I just took the identical PR from master. #8411 
I just need to find out the syntax, which one would you suggest to use the trusty ?

File: .travis.yml [hunk 1]
```diff
@@ -123,8 +101,6 @@ matrix:
         - os: linux
           language: python
           python: 3.7
-          dist: xenial    # required for Python 3.7 (travis-ci/travis-ci#9069)
-          sudo: required  # required for Python 3.7 (travis-ci/travis-ci#9069)
```
Comment (levitte) [inline .travis.yml]: Regarding adding another dist to the matrix, you can see the syntax here :wink:

**Comment by levitte** [comment]
> which one would you suggest to use the trusty ?

Haven't really looked.  Would you mind suggesting something?

**Comment by bernd-edlinger** [comment]
Difficult.  It should be something with --strict-warnings,
Maybe I pick one of he clang builds, they seem nearly identical,
compiler: clang
env: CONFIG_OPTS="--strict-warnings -D__NO_STRING_INLINES"

and 
compiler: clang
env: CONFIG_OPTS="--strict-warnings -D__NO_STRING_INLINES no-deprecated" BUILDONLY="yes"

**Comment by levitte** [comment]
That does seem suitable enough, yes!

**Comment by bernd-edlinger** [comment]
@levitte do you intend to add the enable-buildtest-c++ option to 1.1.1 ?
I removed the tests for this configuration, the build still fails because g++ is used incorrectly

**Comment by bernd-edlinger** [comment]
I can of course always specify the matching g++ compiler, but it seems to be the wrong way of fix.
Maybe simply revert the change in 1.1.1 that introduce this new dependency?

**Comment by levitte** [comment]
> @levitte do you intend to add the enable-buildtest-c++ option to 1.1.1 ?

No

**Comment by bernd-edlinger** [comment]
Okay, so would you like to revert the changes that made the build use the C++ compiler?
Or should I set the CXX environment variable to work around the issue?
```
g++  -Iinclude -std=c++11 -pthread -m64 -fsanitize=memory -fno-omit-frame-pointer -g -Wall -O3 -Wno-unused-command-line-argument -DNDEBUG -D__NO_STRING_INLINES -MMD -MF test/buildtest_aes_cc.d.tmp -MT test/buildtest_aes_cc.o -c -o test/buildtest_aes_cc.o test/buildtest_aes.cc
[01m[Kg++:[m[K [01;31m[Kerror: [m[Kunrecognized argument to -fsanitize= option: '[01m[Kmemory[m[K'
Makefile:8159: recipe for target 'test/buildtest_aes_cc.o' failed
make[1]: *** [test/buildtest_aes_cc.o] Error 1
make[1]: Leaving directory '/home/travis/build/openssl/openssl'
Makefile:203: recipe for target 'tests' failed
make: *** [tests] Error 2
```

**Comment by bernd-edlinger** [comment]
@levitte are you aware that currently 3 out of 19 1.1.1 travis builds configs are failing,
(with this patch still 1 out of 19) because the C++ compiler is used from GCC and
does therefore not understand clang flags in some cases?
I consider that a breaking change in the 1.1.1 version (happened short after 1.1.1b)
in master this was fixed by adding enable-buildtest-c++ config option and not making
it the default.

**Pull request closed without merging**